src/main/java/appeng/core/sync/BasePacket.java

```diff
@@ -32,6 +32,14 @@
 import appeng.core.sync.network.NetworkHandler;
 
 public abstract class BasePacket {
+
+    /**
+     * Sadly {@link PacketBuffer#readString()} gets inlined by Proguard which means it's not available on the
+     * Server. This field has the default string length that is used for writeString, which then also should
+     * be used for readString when it has no special length requirements.
+     */
+    public static final int MAX_STRING_LENGTH = 32767;
```

Comment:
I would actually consider reducing this to a more sensible amount. This will actually allocate a buffer of about 131kb for each string. Something in the range of 20-100 chars is probably more than enough for us.

Reply:
There is one that synchronizes JSON based data between containers. So I was hesistent. The default for WriteString() is 32k too

Reply:
oh, also: no, the buffer is not allocated based on this value, this is just a malicious client security feature.